### PR TITLE
RhysTyne_s46481894_COMP3710_ViT_v2

### PR DESCRIPTION
**Pull request by notkrieger**
Rhys Tyne's COMP3710 report, trying to solve problem 6: ADNI classification using a visual transformer. 
model produced a testing accuracy of 0.68.

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: 
- Need more details on how ViT works to show your understanding of transformer and attention mechanisms.
- Only the val accuracy plot is shown. No training accuracy and validation accuracy.
- Need to compare training acc with val acc to tell how bad the overfitting is


Commit messages: good

Code:
- Uses Vision transformer
- Well commented
- Good design

Functionality/Performance:
- No patient-level split
- Model is overfitting
- Test acc is 68.12%


General comments:
**PR is in the main branch**

- Solves the problem appropriately although performance is poor

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem but no visual results to show training plots -2
Driver Script present
File structure present, but some files incorrect directory (in parent directory instead of folder) -2
No Usage & Demo & Visualisation & Data usage -1
Module present
Commenting
No Data leakage, but patient split required -1
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable
Model/technical explanation minimal -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in InCorrect Branch) -2
Feedback required, files need to be moved into directory and repo README restored.
Request Description minimal -1

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.
